omegatrade-app/backend/app/controller/user.controller.js

```diff
@@ -112,33 +74,28 @@ exports.login = function (req, res) {
  * 
  * @param {*} req contains request headers and payload businessEmail,password
  */
-exports.getToken = function (req, res) {
-    const body = req.body;
-    if (body && body.provider === 'GOOGLE') {
-        User.findUser(body.email, function (err, user) {
+exports.getToken = async function (req, res) {
+    try {
+        const body = req.body;
+        if (body && body.provider === 'GOOGLE') {
+            const [user] = await User.findUser(body.email)
             if (user && user.userId) {
-                    delete user.password;
-                    const token = jwt.sign(user, process.env.JWT_SECRET, {
-                        expiresIn: process.env.EXPIRE_IN
-                    });
-                    res.status(200).json({
-                        success: true,
-                        message: 'Logged in successfully',
-                        userInfo: user,
-                        authToken: token
-                    });
-                
-            } else
-                res.status(401).json({
-                    success: false,
-                    message: 'please register your account',
-                    redirect: 'sign-up'
+                delete user.password;
+                if (user.provider === '' || user.provider === null) {
```

Comment:
I assume that this function is intended to link an existing user with a Google account, and that Google is currently the only provider that is supported, but I even then I think this function should take into account that an existing user might have another provider registered. If there is a user in the database with a provider that is not null or empty, but also not `Google`, then this function will still accept that user and then change the provider from `Something` to `Google`. That doesn't seem right. Or am I misunderstanding the working of this function?

Reply:
yes, This Function is mainly intended to link an `existing normally registered user` with google account.
currently in frontend we are allowing to register `normal user`  and `Google User` only.
I've assumed that we will have only these two types of user, so didn't created another table to store multiple providers.

But yes, I agreed to your point that it will replace any other provider. 
In order to solve this issue,I've to create another schema of `example : social-users table` to store multiple providers and link with it main user. please suggest.

Reply:
If `Google` is the only provider that is supported, then I would suggest adding a simple `else if` statement to the block that verifies that the current provider of the user is actually `GOOGLE`. So something like this:

```typescript
if (user.provider === '' || user.provider === null) {
  ...
} else if (user.provider !== 'GOOGLE') {
  throw Error('The user is not a Google user');
}
```